Test CI Failure

loay opened the pull request:
Adding package.json to the root folder to test CI

Connect to strongloop-internal/scrum-loopback#674

rmg commented on LoopbackGuideSkeleton/build.gradle:
```diff
@@ -40,4 +40,4 @@ android {
         release.setRoot('build-types/release')
     }
-}
+}
```
You appear to be removing the newline from the end of the file. Please don't do that.

loay replied:
Totally my editor :) I will fix it.

rmg commented on package.json:
```diff
@@ -0,0 +1,26 @@
+  "description": "LoopBack mobile application to show developers how to use LoopBack Open Node.js Mobile API Middle tier.",
+  "main": "index.js",
+  "scripts": {
+    "test": ""
```
This means that all CI is doing is running `npm install`, which just installs the devDependencies listed below..

Are there actual tests to run?

loay replied:
No there are not tests to run. The test module is empty ``` "test": ""  ```  in `package.json`
I will test something now and get back to you on that.

bajtos commented:
IIRC there are no tests to run. We just need to build both gradle projects.

bajtos commented:
To be honest, I think this PR is going in a wrong direction. We should not add a package.json, but instead change Jenkins Job config to explicitly run gradle.

See https://github.com/strongloop-internal/scrum-loopback/issues/674

>  I am proposing to setup the CI job to build the android apps only by running ./gradlew build in both directories LoopbackGuideApplication and LoopbackGuideSkeleton. I think we should use a simplified version of the configuration we have in http://ci.strongloop.com/job/loopback-sdk-android/. Simplified because we don't need to publish build artefacts to Artifactory.

Having said that, I am ok with whatever @rmg approves as good.

loay commented:
it is still going through gradle in both files as indicated in the main index.
I will fix the editor typo and wait for @rmg concerning the jenkins job

loay commented:
I removed the devDependencies so the script won't install them and tests are success. 
when I removed
```
  "scripts": {
    "test": ""
  }
``` 
The tests failed. 
let me know if we need to go for a different direction where the package file has to be removed !

rmg commented:
@loay package.json is just a way of telling CI how to run the tests.. since you aren't telling it, it doesn't know, so it fails.

As @bajtos suggested, this is probably the wrong direction if there are no tests to run. Furthermore, this projects looks like it probably requires a specific CI environment. Most likely a new static job needs to be defined in the ci-infrastructure repo similar to https://github.com/strongloop-internal/ci-infrastructure/blob/master/static/loopback-sdk-android.xml

bajtos commented:
> @loay it is still going through gradle in both files as indicated in the main index.

Gradle is a bash file. When you call `require('path/to/gradle')`, Node will read the content of the file and then discard it, no build is triggered.

Few more notes:
 - This project requires Android SDK to get built, I doubt we have it on all different build slaves.
 - The default `package.json`-based CI config runs each build on a matrix of different platforms and Node versions. We don't need that for this package, it's enough to build it on a single platform against a single Android SDK version.

loay commented:
@bajtos @rmg 
I am on it. Thanks for the tips.

bajtos commented:
I'm closing this PR as we need to take a different direction.

loay commented:
I made another PR to add another xml file for the getting-started android repo. 
I thought we still need a package file as in the sdk-android
